homework/index.js

```diff
@@ -43,8 +90,44 @@
         });
         return;
       }
-      const ul = createAndAppend('ul', root);
-      repos.forEach(repo => renderRepoDetails(repo, ul));
+      const header = createAndAppend('header', root, {
+        class: 'header-class',
+      });
+      createAndAppend('h1', header, {
+        text: 'HYF Repository',
+        class: 'title',
+      });
+      const select = createAndAppend('select', header, {
+        class: 'select-class',
+      });
+      const mainContainer = createAndAppend('section', root, {
+        class: 'main-container',
+      });
+      const reposContainer = createAndAppend('section', mainContainer, {
+        class: 'repos-container',
+      });
+      const contributorContainer = createAndAppend('section', mainContainer, {
+        class: 'contributor-container',
+      });
+      repos
+        .sort((a, b) => {
+          return a.name.localeCompare(b.name);
+        })
+        .map((elem, index) => {
+          createAndAppend('option', select, {
+            value: index,
+            text: repos[index].name,
+          });
+        });
+
+      select.addEventListener('change', () => {
+        reposContainer.innerHTML = '';
+        contributorContainer.innerHTML = '';
+        const repo = repos[select.value];
+        renderRepoDetails(repo, reposContainer);
+        contributorContainer.innerHTML = 'Contributors';
+        renderContributors(repo.contributors_url, contributorContainer);
+      });
     });
```

Comment:
You are not displaying the details of the first repo at start-up, as required by the assignment. Because the `change` event only fires when the user actively changes the selection in select box, but not at start-up, you need run the same code that you have in the event handler once, in function main, outside of the event handler, using the repository at index zero:

```js
select.addEventListener('change', () => {
  reposContainer.innerHTML = '';
  contributorContainer.innerHTML = '';
  const repo = repos[select.value];
  renderRepoDetails(repo, reposContainer);
  contributorContainer.innerHTML = 'Contributors';
  renderContributors(repo.contributors_url, contributorContainer);
});

reposContainer.innerHTML = '';
contributorContainer.innerHTML = '';
const repo = repos[0];
renderRepoDetails(repo, reposContainer);
contributorContainer.innerHTML = 'Contributors';
renderContributors(repo.contributors_url, contributorContainer);
```

And it looks like you could write a function to eliminate this code duplication (make the code DRY).

Reply:
I defined a new function then I called it twice for 0 and the select.value.